pkg/vsphere/extraconfig/encode.go

```diff
@@ -252,6 +253,9 @@ func encodeWithPrefix(src interface{}, prefix string) []types.BaseOptionValue {
 				switch field.Interface().(type) {
 				case time.Time:
 					config = append(config, &types.OptionValue{Key: key, Value: field.Interface().(time.Time).String()})
+				case net.IPNet:
+					n := field.Interface().(net.IPNet)
+					config = append(config, &types.OptionValue{Key: key, Value: n.String()})
```

Comment:
Yup - it's annoying and I'm coming to consider the lack of a structured (with standard naming) serialization mechanism like the java serialization interface a flaw in the language